src/prof.c

```diff
@@ -1705,11 +1710,75 @@ prof_getpid(void) {
 static bool
 prof_dump_maps(bool propagate_err) {
 	bool ret;
-	int mfd;
+	int mfd = -1;
 
 	cassert(config_prof);
 #ifdef __FreeBSD__
-	mfd = prof_open_maps("/proc/curproc/map");
+	int mib[4];
+	size_t sz;
+	char *buf;
+
+	mib[0] = CTL_KERN;
+	mib[1] = KERN_PROC;
+	mib[2] = KERN_PROC_VMMAP;
+	mib[3] = prof_getpid();
+
+	if (prof_dump_write(propagate_err, "\nMAPPED_LIBRARIES:\n") &&
+	    propagate_err) {
+		ret = false;
```

Comment:
Just noticed, this should be `ret = true`.

Reply:
ah right. thx for your patience.

Reply:
Thank you for bearing with all the nitpicks :)